tools/c7n_kube/c7n_kube/actions/labels.py

```diff
@@ -56,3 +59,91 @@ def register_resources(klass, registry, resource_class):
         model = resource_class.resource_type
         if hasattr(model, 'patch') and hasattr(model, 'namespaced'):
             resource_class.action_registry.register('label', klass)
+
+
+class EventLabelAction(EventAction):
+    """
+    Label a resource on event
+
+    .. code-block:: yaml
+
+        policies:
+          - name: 'label-foo-on-creation'
+            resource: 'k8s.deployment'
+            mode:
+              type: k8s-admission
+              on-match: allow
+              operations:
+                - CREATE
+            actions:
+              - type: event-label
+                labels:
+                  foo: bar
+    """
+
+    schema = type_schema('event-label', labels={'type': 'object'}, required=['labels'])
+
```

Comment:
since its admission controller mode specific, this could use a validate method to inspect the mode and assert that.